core/arch/arm/plat-stm32mp1/main.c

```diff
@@ -584,3 +585,128 @@ early_init_late(init_debug);
 
 /* Some generic resources need to be unpaged */
 DECLARE_KEEP_PAGER(pinctrl_apply_state);
+
+#if TRACE_LEVEL >= TRACE_DEBUG
+static const char *const dump_table[] = {
+	"usr_sp",
+	"usr_lr",
+	"irq_spsr",
+	"irq_sp",
+	"irq_lr",
+	"fiq_spsr",
+	"fiq_sp",
+	"fiq_lr",
+	"svc_spsr",
+	"svc_sp",
+	"svc_lr",
+	"abt_spsr",
+	"abt_sp",
+	"abt_lr",
+	"und_spsr",
+	"und_sp",
+	"und_lr",
+#ifdef CFG_SM_NO_CYCLE_COUNTING
+	"pmcr",
+#endif
+#ifdef CFG_FTRACE_SUPPORT
+	"cntkctl",
+	"pad"
+#endif
+};
+
+static void stm32mp_dump_core_registers(bool panicking)
+{
+	struct sm_nsec_ctx *sm_nsec_ctx = sm_get_nsec_ctx();
+	static unsigned int lock = SPINLOCK_UNLOCK;
+	bool display = false;
+	uint32_t *reg = NULL;
+	size_t i = U(0);
+
+	cpu_spin_lock(&lock);
+
+	if (panicking)
+		display = true;
+
+	if (!display || !sm_nsec_ctx)
+		return;
+
+	DMSG("CPU : %zu\n", get_core_pos());
+
+	reg = (uint32_t *)&sm_nsec_ctx->ub_regs.usr_sp;
+
+	for (i = U(0); i < ARRAY_SIZE(dump_table); i++)
+		DMSG("%10s : %#8x\n", dump_table[i], reg[i]);
+
+	cpu_spin_unlock(&lock);
+}
+DECLARE_KEEP_PAGER(stm32mp_dump_core_registers);
+#else
+static inline void stm32mp_dump_core_registers(bool panicking __unused) { }
+#endif
+
+#define ARM_CNTXCTL_IMASK	BIT(1)
+
+static void stm32mp_mask_timer(void)
+{
+	/* Mask timer interrupts */
+	write_cntp_ctl(read_cntp_ctl() | ARM_CNTXCTL_IMASK);
+	write_cntv_ctl(read_cntv_ctl() | ARM_CNTXCTL_IMASK);
+}
+
+/* SGI9 (secure SGI 1) informs targeted CPU it shall reset */
```

Comment:
Reported in https://github.com/OP-TEE/optee_os/pull/6412#discussion_r1377198941_

Reply:
Done